### PR TITLE
chore: add documentation scaffolding

### PR DESCRIPTION
**Pull request by gruyaume**
# Description

Here we add scaffolding for the Charmed OAI RAN documentation.

This includes:
- Github issue template
- Github CI
- README
- License
- Sphinx scaffolding
- Code owners

This does not include any content at the moment.

### CODE REVIEW

File: .github/workflows/automatic-check.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+name: Automatic documentation checks
+
+on:
+  - push
+  - pull_request
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
```
Comment (gatici) [inline .github/workflows/automatic-check.yml:8]:
```suggestion
  group: ${{ format('{0}-{1}', github.workflow, github.ref_name) }}
```

Comment (gatici) [reply .github/workflows/automatic-check.yml:8]: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#format is used here.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Contributing
+
+## Build and test the documentation
+
+```bash
+cd docs
+make install
```
Comment (gatici) [inline CONTRIBUTING.md:7]: If we run `make install; make run`  _build folder does not exists and following error appears:
```
make run
. .sphinx/venv/bin/activate; sphinx-autobuild -c . -b dirhtml "." "_build"
Traceback (most recent call last):
  File "/home/gatici/charmed5g/charmed-oai-ran/docs/.sphinx/venv/bin/sphinx-autobuild", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/gatici/charmed5g/charmed-oai-ran/docs/.sphinx/venv/lib/python3.11/site-packages/sphinx_autobuild/__main__.py", line 34, in main
    args, build_args = _parse_args(list(argv))
                       ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gatici/charmed5g/charmed-oai-ran/docs/.sphinx/venv/lib/python3.11/site-packages/sphinx_autobuild/__main__.py", line 101, in _parse_args
    args.outdir = Path(sphinx_args.outputdir).resolve(strict=True)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/pathlib.py", line 994, in resolve
    s = os.path.realpath(self, strict=strict)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen posixpath>", line 416, in realpath
  File "<frozen posixpath>", line 451, in _joinrealpath
FileNotFoundError: [Errno 2] No such file or directory: '_build'
make: *** [Makefile:33: run] Error 1
```
Instead if we run `make install; make html; make run` the page is served.

**Review by gatici — CHANGES_REQUESTED** [review]

File: .github/workflows/automatic-check.yml (hunk 1 above)
Comment (gruyaume) [reply .github/workflows/automatic-check.yml:8]: Here I was using the same as in Charmed Aether SD-Core documentation

File: CONTRIBUTING.md (hunk 2 above)
Comment (gruyaume) [reply CONTRIBUTING.md:7]: Should be fixed now

**Review by gatici — APPROVED** [review]

**Pull request merged**